### PR TITLE
Support deleting SKUs and products.

### PR DESCRIPTION
**Pull request by rasmusrygaard**
r? @brandur 
cc @jimdanz 

API endpoints: [SKU deletion](https://stripe.com/docs/api#delete_sku), [product deletion](https://stripe.com/docs/api#delete_product)

### CODE REVIEW

File: product/client_test.go [hunk 1]
```diff
@@ -134,7 +134,7 @@ func TestProductUpdate(t *testing.T) {
 
 func TestProductList(t *testing.T) {
 	const runs = 3
-	randID := fmt.Sprintf("example.com/", RandSeq(16))
+	randID := fmt.Sprintf("example.com/%s", RandSeq(16))
```
Comment (brandur) [inline product/client_test.go]: Oops, thanks for the fix!

**Comment by brandur** [comment]
Looks good to me @rasmusrygaard!

Just as a general sanity check: can you confirm that the new tests looked okay when running locally?

**Comment by rasmusrygaard** [comment]
Yup, local tests are looking good!

**Comment by brandur** [comment]
Released as 11.1.0 in https://github.com/stripe/stripe-go/commit/8c57a8d3acee8cbc025e897bcaf2d9c5b50060ac.

**Pull request merged**